### ISSUE TITLE
Elements Single Release

### ISSUE THREAD

**Issue by alexkilgour** [springernature/frontend-toolkits#840]
## What

Update Elements from publishing each component as a separate package to publishing the whole library together.

## Why

Now that Elements is more mature this model of working simplifies many things about the design system and allows a much easier way of standardising across products.

## Done when

- [ ] Remove any old/unused components
- [ ] Update frontend-package-manager
- [ ] Stable design tokens
	- [ ] Discuss the scope of design tokens
- [ ] Allow interdependency on components, any areas for consolidation
	- [ ] e.g create global-buttons
	- [ ] any other cases where we have duplication?
- [ ] Bundle in any major breaking changes to brand-context
	- [ ] e.g see [Absolute design token values used](https://github.com/springernature/frontend-toolkits/pull/823) in this PR
- [ ] Update all components to work with the latest brand-context
- [ ] Provide compiled assets in components (@alexkilgour working on atm)
- [ ] Review open issues in this repository

## Anything else

The above list is currently a work in progress to be discussed. There may be more.

- [article on component versioning](https://sid.st/unpolished/component-versioning/)

**Comment by alexkilgour** [comment]
- Will probably need to "lock" the repo at some point to carry this out which means no updates to components
- May have to maintain older versioned components for a period until everyone is updated

**Comment by sturobson** [comment]
a few thoughts:

Github's Design System (Primer) was originally a single release. In ~2017 it went to a monorepo. In 2019 it started moving back to single release. Today it looks like it's slightly split. They have separate repos for: Tokens, CSS, and Components (in various languages). This is visualised in [a diagram](https://github.com/primer/primitives) in their Design Tokens repo. I mention this as Sid (author of the blogpost listed) works at Github now. 

This is similar to how [Wikipedia do it also](https://github.com/wikimedia/codex/tree/main/packages). They have a looser monorepo making use of noms workspaces.

Both of these examples have their own separate GH organisation to keep everything in (nice and tidy). 

Wikimedia go one step further and include their documentation site. I think that's a good thing. I think it would be good for the existing toolkits to have a better tie to elements. 

Thoughts on an "Elements" org? Something like:

elements-themes (brand-context)
elements-tokens (design tokens, currently inside of context)
elements-components (toolkits)
elements-docs (ux-elements-docs, elements-design-system, others)

(I've split out tokens separately as they will hopefully be informing the docs site, and have the potential to work with Figma (if digital design move to Figma) or Sketch (if Sketch opens up their API a little more).

**Comment by sturobson** [comment]
at some point should this be an front-end open space ticket to be discussed?

**Comment by alexkilgour** [comment]
> at some point should this be an front-end open space ticket to be discussed?

Yep, it should be. A number of people are aware this is the overall plan and I don't think we will find opposition to the idea, but would be good to discuss once we have an idea of what needs to be done and what it will look like

**Comment by alexkilgour** [comment]
> Thoughts on an "Elements" org?
This would simplify things from our perspective, but given how SN is managed on github I'm not sure we will be able to do this. Worth having the conversation though

**Comment by sturobson** [comment]
>given how SN is managed on github I'm not sure we will be able to do this.  

perhaps something similar to how Wikimedia handle it? 

github.com/springernature/elements/
github.com/springernature/elements/elements-themes
github.com/springernature/elements/elements-tokens
github.com/springernature/elements/elements-components
github.com/springernature/elements/docs

**Comment by sturobson** [comment]
With a move to a single release, or set of releases I think we should look at the contribution model and move towards a roadmap of work.

**Comment by sturobson** [comment]
another benefit would be a smoother transition to `@use` and `@forward` in Sass (as `@import` is still set for deprecation)

**Comment by amyhupe** [comment]
Reviewed this in planning today - have assigned @sturobson as the owner so that we have one person responsible for tracking and updating this issue
